xarray/tests/test_plot.py

```diff
@@ -3218,3 +3218,19 @@ def test_facetgrid_axes_raises_deprecation_warning():
             ds = xr.tutorial.scatter_example_dataset()
             g = ds.plot.scatter(x="A", y="B", col="x")
             g.axes
+
+
+@requires_matplotlib
+@requires_nc_time_axis
+def test_plot_nc_time() -> None:
+    da = xr.DataArray(
+        range(10),
+        dims="time",
+        coords={
+            "time": xr.cftime_range(
+                "1900-01-01", periods=10, calendar="noleap", freq="D"
+            )
+        },
+    )
+    with figure_context():
+        da.plot()
```

Comment:
So we have tests along these lines (e.g. [here](https://github.com/pydata/xarray/blob/cc7e09a3507fa342b3790b5c109e700fa12f0b17/xarray/tests/test_plot.py#L2825-L2850)).  I wonder if the reason this isn't caught is that the import that takes place in the [`_importorskip("nc_time_axis")` step](https://github.com/pydata/xarray/blob/main/xarray/tests/__init__.py#L71) preemptively registers the cftime date types with matplotlib (so the import that occurs within the plotting code ends up being redundant)?

Reply:
Thanks for looking into this issue by the way.

Reply:
Oh wow, never occured to me, haha. Then also this test doesn't help much.

Maybe one could do some trickery to remove it from sys.modules, but that probably simply breaks the registering and does not undo it.

No idea on how to test this properly then...

Reply:
haha yeah, it clearly didn't occur to me when I was reviewing these tests either.  Maybe another way would be to remove the `_importorskip` and move the skipping logic within the tests themselves, e.g. within a try-except block, skipping the test if an `ImportError` occurred during the plotting call. If we wanted to go a step further we could even check that explicitly importing `nc_time_axis` led to an `ImportError` at that point before skipping just to be sure it wasn't caused by another library.

I agree without some kind of trickery, I don't see a way to be able to use our typical test skipping logic in this case.

Reply:
I ~think~ it should work now.
I removed the importskip and replaced it with a lazy version that only checks if the module is available without importing it.

Reply:
Ah nice -- I like that approach much better.